### ISSUE TITLE
Disappearing HLS_MANIFEST_URL for cloudflare stream?

### ISSUE THREAD

**Issue by khalilsarwari** [Schachte/cloudflare-stream-downloader#1]
Hi,

Thanks for putting together this tool; it seems to me that the hls manifest is no longer accessible for a video after some time has elapsed?

Can you please confirm if this case, and when does it happen if it is the case?

**Comment by Schachte** [comment]
@khalilsarwari Can you elaborate on what behavior you're seeing and how to reproduce it?

**Comment by khalilsarwari** [comment]
Hi @Schachte

This is not an issue with the tool, but I thought perhaps you may have run into it as part of using/testing the tool.

Occasionally, when attempting to fetch from hls manifest url, cloudflare returns a 404 with the message 'Manifest not found'. I have not been able to determine when or why this happens; it does not happen for all videos (these videos are recordings that are saved from a live stream). The issue persists even after turning signed urls off; I have the video uid/url and can share it privately if that helps.

I have already asked on the stream discord as well as in the help center with no response yet, so I though I'd try my luck here.

**Comment by Schachte** [comment]
@khalilsarwari Are you caching the manifest urls on your end by chance? I would first ensure that you're not serving cached manifests to your clients as the manifest content is dynamic. I recommend starting a thread on the Cloudflare Stream forums with some sample video UIDs and the behavior you've described. Feel free to post the thread link here so I can close this issue.

**Comment by khalilsarwari** [comment]
I am not caching the manifest url; I have confirmed the issue by opening the video page in the cloudflare dashboard, copying the hls manifest url from there, and doing a GET with curl and i get Manifest not found. However for other videos when I do this, it works fine.

I have started a thread here: [https://community.cloudflare.com/t/hls-manifest-not-found/586990](https://community.cloudflare.com/t/hls-manifest-not-found/586990)

Thanks for your time!
